src/data/finsupp/basic.lean

```diff
@@ -870,6 +870,13 @@ See `finsupp.lapply` for the stronger version as a linear map. -/
 @[simps apply]
 def apply_add_hom (a : α) : (α →₀ M) →+ M := ⟨λ g, g a, zero_apply, λ _ _, add_apply _ _ _⟩
 
+/-- Coercion from a `finsupp` to a function type is an `add_monoid_hom`. -/
+noncomputable
+def coe_fn_add_monoid_hom {α M : Type*} [add_zero_class M] : (α →₀ M) →+ (α → M) :=
+{ to_fun := λ f, f,
+  map_zero' := by simp,
+  map_add' := by simp }
```

Comment:
You should be able to replace these calls to `simp` with a direct application of `coe_zero` and `coe_add`